apps/omg_api/lib/state/core.ex

```diff
@@ -428,6 +435,67 @@ defmodule OMG.API.State.Core do
     {:ok, {event_triggers, db_updates, validities}, new_state}
   end
 
+  defp sigs_chope(<<>>), do: []
+  defp sigs_chope(<<sig::bytes-size(65), rest::binary>>), do: [sig | sigs_chope(rest)]
+
+  @spec in_flight_exits(in_flight_txs :: [in_flight_exit()], state :: t()) ::
+          {:ok, {[exit_event], [db_update]}, new_state :: t()}
+  def in_flight_exits(in_flight_txs, state) do
+    {db_updates_list_and_events, new_state} =
+      in_flight_txs
+      |> Enum.map_reduce(state, &in_flight_exit/2)
+
+    {event_triggers, db_updates} =
```

Comment:
Taking my comment on function naming and `exit_utxos` further, it looks like you should DRY these a lot, so that after you get a list of UTXO positions to work with, all 3 "versions" (for standard exits, for in flight exit inputs, for piggybacks) use the same DRY code.